support/spray/src/test/scala/sjsonnew/support/spray/CalendarFormatsSpec.scala

```diff
@@ -20,16 +20,59 @@ package support.spray
 import spray.json.{ JsValue, JsNumber, JsString, JsNull, JsTrue, JsFalse, JsObject }
 import org.specs2.mutable._
 import java.util.{ Calendar, TimeZone }
+import java.time._
 
 class CalendarFormatsSpec extends Specification with BasicJsonProtocol {
   "The dateStringIso" should {
+    // JDK 8 / Joda dates
+    val zdt = ZonedDateTime.of(1999, 1, 1, 0, 0, 0, 0, ZoneId.of("UTC"))
+    val millis = zdt.plusNanos(1000 * 1000)
+    // zoneless instant in time
+    val inst = millis.toInstant
+    val ld = LocalDate.of(1999, 1, 1)
+
+    // legacy dates
     val seconds = Calendar.getInstance(TimeZone.getTimeZone("UTC"))
     seconds.clear
     seconds.set(1999, 1, 1, 0, 0, 0)
+    
     val milliseconds = Calendar.getInstance(TimeZone.getTimeZone("UTC"))
     milliseconds.clear
     milliseconds.set(1999, 1, 1, 0, 0, 0)
     milliseconds.set(Calendar.MILLISECOND, 1)
+
+    "convert a ZonedDateTime to JsString" in {
+      Converter.toJsonUnsafe(zdt) mustEqual JsString("1999-01-01T00:00:00Z")
+    }
+
+    "convert the JsString back to the ZonedDateTime" in {
+      Converter.fromJsonUnsafe[ZonedDateTime](JsString("1999-01-01T00:00:00Z")).toInstant mustEqual zdt.toInstant
```

Comment:
There's a minor loss of information without converting to `Instant`:

```
[error]   x convert the JsString back to the ZonedDateTime with milliseconds
[error]    '1999-01-01T00:00:00.001Z' is not equal to '1999-01-01T00:00:00.001Z[UTC]' (CalendarFormatsSpec.scala:55)
[error] Actual:   ....001Z[]
[error] Expected: ....001Z[UTC]
```

Technically you could have a situation where timezone offset alone is not enough to cover local summer time rules etc, but I think String representation of a date should stick to ISO 8601 (https://en.wikipedia.org/wiki/ISO_8601), which was also used as part of XML Schema Part 2 (https://www.w3.org/TR/2001/REC-xmlschema-2-20010502/#dateTime)